packages/ttypescript/src/loadTypescript.ts

```diff
@@ -1,32 +1,66 @@
-import * as fs from 'fs';
-import * as resolve from 'resolve';
 import * as TS from 'typescript';
 type ts = typeof TS;
+import { readFileSync } from 'fs';
+import { sync as resolveSync } from 'resolve';
 import { patchCreateProgram } from './patchCreateProgram';
 import { dirname } from 'path';
 import { runInThisContext } from 'vm'
+import Module = require('module')
 
+type TypeScriptModuleFactory = (exports: ts, require: NodeRequireFunction, module: { exports: any }, filename: string, dirname: string) => ts;
+const typeScriptModuleFactoryCache: { [key: string]: TypeScriptModuleFactory } = Object.create(null)
 
 export function loadTypeScript(
     filename: string,
     { folder = __dirname, forceConfigLoad = false }: { folder?: string; forceConfigLoad?: boolean } = {}
 ): ts {
-    const opts = { basedir: folder };
-    const typescriptFilename = resolve.sync('typescript/lib/' + filename, opts);
-
-    const m = { exports: {} as ts }
-    const code = fs.readFileSync(typescriptFilename, 'utf8');
-    runInThisContext(
-        `(function (exports, require, module, __filename, __dirname) {${code}\n});`,
-        { filename: typescriptFilename, lineOffset: 0, displayErrors: true }
-    ).call(m.exports, m.exports, require, m, typescriptFilename, dirname(typescriptFilename));
-    
+    const libFilename = resolveSync('typescript/lib/' + filename, { basedir: folder })
+
+    let factory = typeScriptModuleFactoryCache[libFilename];
+    if (!factory) {
+        const code = readFileSync(libFilename, 'utf8');
+        factory = runInThisContext(
```

Comment:
factory: TypeScriptModuleFactory

Reply:
It is implicitly of type TypeScriptModuleFactory | undefined, but I will make it explicit

Reply:
Ah, it's autoinfered. Ok